.github/workflows/nightly_aws_operational_procedure.yml

```diff
@@ -8,110 +8,235 @@ on:
+      timeout-minutes: 15
       env:
-        HELM_CHART_VERSION: "9.3.3"
-        BACKUP_NAME: nightly-9.3
+        # Pick a known namespace for cross cluster testing
+        CLUSTER_0_NAMESPACE: "snapshot-cluster-0"
+        CLUSTER_0_NAMESPACE_FAILOVER: "snapshot-cluster-0-failover"
+        CLUSTER_1_NAMESPACE: "snapshot-cluster-1"
+        CLUSTER_1_NAMESPACE_FAILOVER: "snapshot-cluster-1-failover"
       run: |
-        go test --count=1 -v -timeout 45m ./multi_region_aws_operational_procedure_test.go -run TestAWSOperationalProcedure
+        go test --count=1 -v -timeout 44m -run TestAWSDNSChaining
+    - name: Remove profile credentials from ~/.aws/credentials
+      if: always()
+      run: |
+        rm -rf ~/.aws/credentials
+
+  operational-procedure:
+    runs-on: ubuntu-latest
+    timeout-minutes: 120
+    needs:
+    - cluster-creation
+    - cluster-configuration
+    strategy:
+      fail-fast: false
+      matrix:
+        c8-version:
+        - 8.3.11
+        - 9.3.3
+        - 10.0.2
+        - SNAPSHOT
 
-    - name: Operational Procedure Test in 2 Regions - C8.5
+    steps:
+    ################## Checkout ##################
+    - name: Checkout repository
+      uses: actions/checkout@v4
+    ############# Tool Installation ##############
+    - name: Setup AWS and Tools
+      uses: ./.github/actions/setup-aws
+      with:
+        secrets: ${{ toJSON(secrets) }}
+    ########### KubeConfig Generation ############
+    - name: Export Cluster Name
+      run: |
+        echo "CLUSTER_NAME=${{ needs.cluster-creation.outputs.cluster_name }}" >> "$GITHUB_ENV"
+    - name: KubeConfig generation
+      working-directory: ./test
+      timeout-minutes: 5
+      run: |
+        go test --count=1 -v -timeout 4m -run TestAWSKubeConfigCreation
+    #### Export required environment variables ####
+    - name: Export C8 namespaces and versions
+      run: |
+        version=${{ matrix.c8-version }}
+        version_with_hyphens="${version//./-}"
+
+        if [ "$version" == "SNAPSHOT" ]; then
+          {
+            echo "GLOBAL_IMAGE_TAG=SNAPSHOT"
+            echo "HELM_CHART_VERSION=10.0.2"
+            version_with_hyphens="snapshot"
+          } >> "$GITHUB_ENV"
+        else
+          echo "HELM_CHART_VERSION=${version}" >> "$GITHUB_ENV"
+        fi
+
+        {
+          echo "CLUSTER_0_NAMESPACE=${version_with_hyphens}-cluster-0"
+          echo "CLUSTER_0_NAMESPACE_FAILOVER=${version_with_hyphens}-cluster-0-failover"
+          echo "CLUSTER_1_NAMESPACE=${version_with_hyphens}-cluster-1"
+          echo "CLUSTER_1_NAMESPACE_FAILOVER=${version_with_hyphens}-cluster-1-failover"
+          echo "BACKUP_NAME=nightly-${version_with_hyphens}"
+        } >> "$GITHUB_ENV"
+    ########### Operational Procedure ############
+    - name: Deploy - ${{ matrix.c8-version }}
+      working-directory: ./test
+      timeout-minutes: 21
+      run: |
+        go test --count=1 -v -timeout 20m -run TestAWSDeployDualRegCamunda
```

Comment:
everything has to be done sequentially as everything is relying on the step prior, so there's no possibility to run things in parallel within go.
You can't deploy C8 without having set up the DNS chaining, you can't do the failover without C8 deployed, ...
That's why logical abstractions are grouped into tests to easily identify what's going on.

If I had to run it locally, the same way as CI, sequentially test group by test group.

Reply:
thanks for elaborating on that approach!
I think I'll keep it as it is for now since I don't see a big advantage in this as I still need everything sequentially and still things to be separated to profit of it in GHA.